addon/decorators/changeset-route.js

```diff
@@ -0,0 +1,22 @@
+import { assert } from '@ember/debug';
+import { isNone } from '@ember/utils';
+import createChangeset from './create-changeset';
+
+export default function changesetRoute(RouteSubclass) {
+    class ChangesetRoute extends RouteSubclass {
+        setupController(controller, model) {
+            this._super(...arguments);
```

Comment:
I _think_ this needs to be `super(...arguments);` here, to follow native class syntax

Reply:
@billdami I tried that first, and super/super.setupController didn't exist

Reply:
@bakerac4 i wonder if because in your test the `Route` class you tested it against was an old-style Ember.Object based route instance, i.e. the dummy app is not running on the new ember octane core.